utils/zerovec/src/varzerovec/vec.rs

```diff
@@ -451,8 +455,14 @@ where
 {
     #[inline]
     fn eq(&self, other: &VarZeroVec<'b, T, F>) -> bool {
-        // VarULE has an API guarantee that this is equivalent
-        // to `T::VarULE::eq()`
+        // VZV::from_elements used to produce a non-canonical representation of the
```

Comment:
though this is ok too, note that this also breaks the VarULE "equality is byte equality" invariant on VZS

Reply:
if we remove this we should also special case is_empty to check for empty length

Reply:
`is_empty` should already work, because it checks' VZComponent's indices slice, which gets special-case initialised.

> note that this also breaks the VarULE "equality is byte equality" invariant on VZS

We already broke that invariant

Reply:
Note that testdata CI is failing so the incorrect representation is out there